src/Sarif.Multitool.FunctionalTests/ValidateCommandTests.cs

```diff
@@ -10,13 +10,13 @@ namespace Microsoft.CodeAnalysis.Sarif.Multitool
 {
     public class ValidateCommandTests : SarifMultitoolTestBase
     {
-        [Fact(DisplayName = nameof(ValidateCommand_ReportsJsonSyntaxError), Skip = "JSchema requires an update.")]
+        [Fact(Skip = "JSchema attempts to instantiate types from the compiled Sarif assembly that no longer exist")]
```

Comment:
>JSchema [](start = 22, length = 7)

Here's what's happening: when the validation code loads during testing, our tested/in-flight SARIF.DLL is already loaded in the default app domain. JSchema therefore attempts to load Rule, IRule, RuleConfiguration and Result objects from it. it is very interesting that JSchema has been effectively upgrade to current SARIF dll without breaking previously, I was originally able to work around this problem by reproducing the rule factory code from JSchema (that builds a rule object from the jschema error identifiers). At some point, however, the divergence of the result objects broke as well. Easiest path forward: update Jschema then reenable all these tests.

Reply:
You know, JSchema _hasn't_ always upgraded successfully from one version to the next. That's why we had that long discussion about refactoring the packages and building them in a particular order -- first JSchema + JSchemaToDotNet, then SARIF SDK, then JSchema validator, then the rest of the SARIF packages. And when I got around to doing that, the problem we'd observed -- whatever it was, who knows 18 months later -- was gone and I never bothered.

Yes, we can work around the problem again by disabling this tests, upgrading JSchema, and re-enabling the tests.

---
In reply to: [255250496](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255250496) [](ancestors = 255250496)

Reply:
Really good explanation, BTW.

---
In reply to: [255250496](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255250496) [](ancestors = 255250496)